framework/src/base/FEProblemBase.C

```diff
@@ -4401,6 +4399,25 @@ FEProblemBase::possiblyRebuildGeomSearchPatches()
     {
       case 0: // Never
         break;
+      case 3: // Nonlinear iteration
```

Comment:
Minor: This should really be cleaned up. (i.e. MooseEnums should be used locally within a class but not passed around so much because they can get out of sync easily with their integer values). A better approach would be to create a new global enumeration that you dump the MooseEnum into and then actually use the real enum here. This doesn't have to be part of this PR.  #10349

Reply:
I have added a new PatchUpdatetype to MooseTypes and I am using that instead of the unsigned ints in the other files.